index.js

```diff
@@ -374,6 +376,36 @@
     return Nothing();
   });
 
+  //. indexOf :: a -> [a] -> Maybe Number
+  //.
+  //. Returns Just the position of the first occurrence of an
+  //. element in a list, or Nothing if the element does not occur
+  //. in the list.
+  //.
+  //. ```javascript
+  //. > S.indexOf('c', ['a','b', 'c', 'd', 'e'])
+  //. Just(2)
+  //. ```
+  S.indexOf = R.curry(function(target, list) {
+    var idx = R.indexOf(target, list);
+    return indexOfToMaybe(idx);
+  });
```

Comment:
Nice. But I don't think it gets us of the hook, this is a pattern that's going to be repeated in the `indexOf` methods for the strings so it'll still make sense to have a function that converts any function where a -1 indicates a `null` value and reuse it across the board.